--- a/etc/init.d/omcid.sh	2021-09-17 21:27:01.000000000 -0500
+++ b/etc/init.d/omcid.sh	2023-03-17 10:39:11.000000000 -0500
@@ -41,22 +41,31 @@
 	local omci_iop_mask
 	local lct=""
 
-	#is_flash_boot && wait_for_jffs
-
 	config_load omci
+	
+	tmp=`fw_printenv omci_hw_ver 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		sed "s/256 0 HWTC 0000000000000/256 0 HWTC ${tmp}/" /rom/etc/mibs/data_1g_8q_us1280_ds512.ini > /tmp/auto_generated_mib.ini
+		mib_file="/tmp/auto_generated_mib.ini"
+	else
+		mib_file="/rom/etc/mibs/data_1g_8q_us1280_ds512.ini"
+	fi
 
-	tmp=`fw_printenv mib_file 2>&- | cut -f2 -d=`
+	tmp=`fw_printenv mib_file_custom 2>&- | cut -f2 -d=`
 	if [ -f "/etc/mibs/$tmp" ]; then
 		mib_file="/etc/mibs/$tmp"
+	fi
+	
+	tmp=`fw_printenv omci_omcc_ver 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		omcc_version=$tmp
 	else
-		config_get mib_file "default" "mib_file" "/etc/mibs/data_1g_8q.ini"
+		omcc_version=160
 	fi
 
 	config_get tmp "default" "status_file" "/tmp/omci_status"
 	status_entry_create "$tmp"
 
-	config_get omcc_version "default" "omcc_version" 160
-
 	case $(uci -q get network.lct.ifname) in
 	lct0)
 		lct=-g1
diff -ru a/etc/init.d/onu.sh b/etc/init.d/onu.sh
