--- a/etc/init.d/sfp_eeprom.sh	2021-09-17 21:27:09.000000000 -0500
+++ b/etc/init.d/sfp_eeprom.sh	2023-03-17 13:18:36.000000000 -0500
@@ -100,16 +100,56 @@
 
 	# reset to default values
 	sfp_i2c -d yes
-	#vendor_config
-	#serialnumber_config
-	#bitrate_config
-	# configure I2C EEPROM addresses
-	#eeprom_addr_config
-	# activate write protection
-	#sfp_i2c -i 0 -l 128 -p 1
-	#sfp_i2c -i 256 -l 128 -p 1
-	# activate write protection for dedicated fields
-	#sfp_i2c -i 366 -p 2 -m 0x87
+	
+	tmp=`fw_printenv sfp_vendor_name 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i0 -s "$tmp"
+	else
+		/opt/lantiq/bin/sfp_i2c -i0 -s "Lantiq"
+	fi
+	
+	tmp=`fw_printenv sfp_part_name 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i1 -s "$tmp"
+	else
+		/opt/lantiq/bin/sfp_i2c -i1 -s "Falcon SFP"
+	fi
+	
+	tmp=`fw_printenv sfp_vendor_rev 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i2 -s "$tmp"
+	else
+		/opt/lantiq/bin/sfp_i2c -i2 -s "0"
+	fi
+	
+	tmp=`fw_printenv sfp_part_serial 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i3 -s "$tmp"
+	fi
+	
+	tmp=`fw_printenv sfp_date_code 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i4 -s "$tmp"
+	fi
+	
+	tmp=`fw_printenv sfp_vendor_data 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i5 -s "$tmp"
+	fi
+	
+	tmp=`fw_printenv omci_equip_id 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i6 -s "$tmp"
+	else
+		/opt/lantiq/bin/sfp_i2c -i6 -s "MA5671B"
+	fi
+	
+	tmp=`fw_printenv omci_vendor_id 2>&- | cut -f2 -d=`
+	if [ -n "$tmp" ]; then
+		/opt/lantiq/bin/sfp_i2c -i7 -s "$tmp"
+	else
+		/opt/lantiq/bin/sfp_i2c -i7 -s "SPTC"
+	fi
 
 	config_get eeprom default eeprom 1
 
diff -ru a/etc/init.d/telnet b/etc/init.d/telnet
